hail/python/hail/linalg/utils/misc.py

```diff
@@ -166,6 +166,16 @@ def locus_windows(locus_expr, radius, coord_expr=None, _localize=True):
     if radius < 0:
         raise ValueError(f"locus_windows: 'radius' must be non-negative, found {radius}")
     check_row_indexed('locus_windows', locus_expr)
+
+    # check loci are in sorted order
+    loci = locus_expr.global_position().collect()
```

Comment:
Wait, is this going from hail to Python back to hail? Isn't just asserting that it's keyed by locus_expr enough?

Reply:
ah, crap. That should be a hl.agg.collect, probably. It's not quite asserting that it's keyed by locus_expr--as long as it's not out-of-order. Probably the same in the vast majority of cases.

Reply:
Yeah, I don't think I mind that constraint, but it's up to you. Alternatively, there's no `KeyBy(die_if_unsorted)` is there?

Reply:
No, there isn't :(. I'd prefer to keep it this way so there's no worry about accidentally suddenly breaking someone's pipeline.

Reply:
I folded it into one aggregation pass, though, so hopefully not too much overhead.